File: frontend/src/types/portfolio.ts
export interface Portfolio {
  portfolio_id: string;
  user_id: string;
  symbol: string;
  quantity: number;
  average_price: number;
  purchase_date?: string;
  created_at: string;
  updated_at?: string;
}

export interface PortfolioWithMarketData extends Portfolio {
  current_price?: number;
  current_value?: number;
  unrealized_pnl?: number;
  unrealized_pnl_percent?: number;
}

export interface CreatePortfolioItem {
  symbol: string;
  quantity: number;
  average_price: number;
  purchase_date?: string;
}

export interface UpdatePortfolioItem {
  quantity?: number;
  average_price?: number;
  purchase_date?: string;
}

File: backend/app/schemas/portfolio.py
from pydantic import BaseModel, UUID4
from typing import Optional
from datetime import date, datetime
from decimal import Decimal

class PortfolioBase(BaseModel):
    symbol: str
    quantity: Decimal
    average_price: Decimal
    purchase_date: Optional[date] = None

class PortfolioCreate(PortfolioBase):
    pass

class PortfolioUpdate(BaseModel):
    quantity: Optional[Decimal] = None
    average_price: Optional[Decimal] = None
    purchase_date: Optional[date] = None

class PortfolioResponse(PortfolioBase):
    portfolio_id: UUID4
    user_id: UUID4
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class PortfolioWithMarketData(PortfolioResponse):
    """Portfolio with current market data"""
    current_price: Optional[Decimal] = None
    current_value: Optional[Decimal] = None
    unrealized_pnl: Optional[Decimal] = None
    unrealized_pnl_percent: Optional[Decimal] = None